src/main/java/com/google/firebase/messaging/LightSettings.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.messaging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.client.util.Key;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A class representing light settings in an Android Notification.
+ */
+public final class LightSettings {
+  
+  @Key("color")
+  private final Color color;
+  
+  @Key("light_on_duration")
+  private final String lightOnDuration;
+  
+  @Key("light_off_duration")
+  private final String lightOffDuration;
+  
+  private LightSettings(Builder builder) {
+    this.color = builder.color;
+    this.lightOnDuration = builder.lightOnDuration;
+    this.lightOffDuration = builder.lightOffDuration;
+  }
+  
+  /**
+   * Creates a new {@link LightSettings.Builder}.
+   *
+   * @return A {@link LightSettings.Builder} instance.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+  
+  public static class Builder {
+
+    private Color color;
+    private String lightOnDuration;
+    private String lightOffDuration;
+
+    private Builder() {}
+
+    /**
+     * Sets the color value with a string.   
+     *
+     * @param color Color specified in the {@code #rrggbb} format.
+     * @return This builder.
+     */
+    public Builder setColorFromString(String color) {
+      this.color = Color.fromString(color);
+      return this;
+    }
+
+    /**
+     * Sets the color value in the light settings.   
+     *
+     * @param color Color to be used in the light settings.
+     * @return This builder.
+     */
+    public Builder setColorFromString(Color color) {
```

Comment:
Minor thing. Should this function be named just `setColor()` instead?

Reply:
Yes. Nice catch!